File: Java/Mit Polymorphismus/HaustierFactory.java
@SuppressWarnings("ALL")
public class HaustierFactory {
    public static Haustier erstelleHaustier(HaustierArt art, String name){
        switch(art){
            case HUND:
                return new Hund(name);
            case KATZE:
                return new Katze(name);
            case WELLENSITTICH:
                return new Wellensittich(name);
            default:
                return new Hund("");
        }
    }
}

File: TypeScript/Mit Polymorphismus/HaustierFactory.ts
import {Haustier} from "./Haustier";
import {HaustierArt} from "./HaustierArt";
import {Hund} from "./Hund";
import {Katze} from "./Katze";
import {Wellensittich} from "./Wellensittich";

class HaustierFactory{
    constructor(){
    }

    static erstelleHaustier(art : number, name : string) : Haustier{
        switch(art){
            case HaustierArt.HUND:
                return new Hund(name);
            case 2:
                return new Katze(name);
            case 3:
                return new Wellensittich(name);
            default:
                return new Hund("");
        }
    }
}

export {HaustierFactory};